nsxt/resource_nsxt_node_user.go

```diff
@@ -212,7 +212,7 @@ func resourceNsxtNodeUserUpdate(d *schema.ResourceData, m interface{}) error {
 	}
 
 	// If password is changed, handle password change.
-	if password != oldPassword {
+	if password != oldPassword && password != "" {
```

Comment:
should the resource error out if password is empty?

Reply:
So what happens is as follows:
When updating in the test [here](https://github.com/vmware/terraform-provider-nsxt/blob/master/nsxt/resource_nsxt_node_user_test.go#L63), password isn't specified at all within the test. This is legit, I think, and therefore the test fails as old_password != new_password.
So the change ignores it - I don't think that the provider should fail in that case.

Reply:
But wouldn't that overwrite the password in the state?

Reply:
Shouldn't we nullify the password on NSX in this case? If empty pwd is allowed?

Reply:
Password cannot be nullified. So maybe the test should be fixed in this case? As the test doesn't set the password to the same as the existing one.

Reply:
My take is that test should be fixed, and possibly validation code added in the resource?